.dotnet/src/Generated/Models/AssistantThread.Serialization.cs

```diff
@@ -43,7 +43,7 @@ protected virtual void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWrit
             }
             if (_additionalBinaryDataProperties?.ContainsKey("metadata") != true)
             {
-                if (Metadata != null && Optional.IsCollectionDefined(Metadata))
+                if (options.Format != "W" && Optional.IsCollectionDefined(Metadata))
```

Comment:
Is this new check for `options.Format != "W"` expected? I see it here and it a few other places.

Reply:
Yes, this looks correct. Metadata is readonly so we should only write if not in wire format.